places/src/api/autocomplete.rs

```diff
@@ -0,0 +1,93 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// Roughly, UnifiedComplete.js's stuff
+
+use url::{Url};
+use error::*;
+use super::connection::{Connection};
+
+// rusqlite imports probably reflect that some of this should be in ::storage
+use rusqlite;
+
+#[derive(Debug)]
+pub struct SearchParams {
+    search_string: String,
+    limit: u32
+}
+
+#[derive(Debug)]
+pub struct SearchResult {
+    title: Option<String>,
```

Comment:
I feel like this should have more information even than this. Like, it probably also needs the row id, but possibly also just the whole place record (sans visits). 

Might be worth looking at the WebExtension history.search API some, although that doesn't use frecency.